pkg/asset/machines/machineconfig/hyperthreading.go

```diff
@@ -0,0 +1,30 @@
+package machineconfig
+
+import (
+	"fmt"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// ForHyperthreading creates the MachineConfig to enable hyperthreading.
+func ForHyperthreading(mode types.HyperthreadingMode, role string) *mcfgv1.MachineConfig {
+	if mode == types.HyperthreadingEnabled {
+		return nil
+	}
+	return &mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("99-%s-hyperthreading", role),
+			Labels: map[string]string{
+				"machineconfiguration.openshift.io/role": role,
+			},
+		},
+		Spec: *specForKarg("nosmt", ""),
```

Comment:
PTAL https://url.corp.redhat.com/0d85a22 for what's expected in terms of the local service. There's a POC there that's turning into a usable service.